elastic/kibana
# [Ingest Manager] Integrate beta messaging with Add Data

Resolves #69048 
Resolves #70384

## Summary

This PR adds Ingest Manager Beta messaging to Kibana's Add Data tutorial directory and tutorial module pages:

- Adds methods to `HomePublicPluginSetup` interface that allows for custom components to be injected into Home > Add data:
  - `tutorial.registerDirectoryNotice()` registers a component that will be displayed on main Add data directory page
  - `tutorial.registerDirectoryHeaderLink()` registers a component that will be displayed on the right of Add data directory page title area
  - `registerModuleNotice()` registers a component that will be displayed under the description of an Add data module tutorial page
- Adds `moduleName?: string;` to `TutorialSchema` and surface the module name of existing tutorials, where applicable (I believe a few tutorials are not associated with modules, such as `EMS Boundaries`)
  - **This field is used to look up whether we have a package with a matching name in Ingest Manager. Most tutorials already had a local `moduleName` variable which I simply surfaced. I attempted to add ones which didn't, and would appreciate reviews for those ones I added.**
- Adds `home` plugin as an optional dependency to Ingest Manager plugin
- Creates Ingest Manager home integration components for all three methods above ^ and, if `home` plugin is available, registers them during Ingest Manager plugin `setup()` lifecycle
- Adds `has_seen_add_data_notice` field to Ingest Manager global settings saved object
  - When Dismiss is clicked in the directory notice, `has_seen_add_data_notice` is set to true, preventing the notice from being seen again on refresh

## Testing

1. Go to Home > Add data, a dismissible callout should be seen about Ingest Manager Beta
2. Click Dismiss, the callout should disappear and a link to Ingest Manager Beta appears on the right of page title
3. Refresh the page, the callout should remain hidden and the link should remain visible
4. Click into a tutorial card for a module that has a package associated with it, such as `AWS Cloudwatch logs` or `MySQL logs`
    - A messaging about Ingest Manager Beta should be shown below the tutorial description
5. Click into a tutorial card for a module that _does not_ have a package associated with it
    - No messaging should be shown about Ingest Manager in that tutorial description

If Ingest Manager is not available as a feature in the current space, no Ingest Manager Beta messaging should appear in Add data.

If you want to get the dismissible callout back, you can clear your setting by:
```
POST /api/ingest_manager/settings
{
  "has_seen_add_data_notice": false
}
```

## Screenshots

![Jul-08-2020 13-27-27](https://user-images.githubusercontent.com/1965714/86966997-db42f700-c11e-11ea-89a4-efd0e4375438.gif)

![image](https://user-images.githubusercontent.com/1965714/86966791-961ec500-c11e-11ea-869c-6eed2f64efb6.png)

![image](https://user-images.githubusercontent.com/1965714/86966732-80a99b00-c11e-11ea-8cae-4369b36786d9.png)

![image](https://user-images.githubusercontent.com/1965714/86966771-8ef7b700-c11e-11ea-9dcf-20bf9e8f12ce.png)
